### PR TITLE
LEARNER-5182: Reuse TieredCache.

### PR DESCRIPTION
**Pull request by robrap**
LEARNER-5182

@nasthagiri: FYI: This is the larger refactor of Ecommerce.  Let me know if you think we want to dedicate my initial time on Arch Squad to trying to use quickcache and skipping this step.  Or, if as discussed, we should land this and return to this when and if we wish to prioritize.

### CODE REVIEW

File: ecommerce/coupons/utils.py [hunk 1]
```diff
@@ -161,8 +162,10 @@ def fetch_course_catalog(site, catalog_id):
             logger.exception("Catalog '%s' not found.", catalog_id)
             raise
 
-    cache.set(cache_key, response, settings.COURSES_API_CACHE_TIMEOUT)
```
Comment (robrap) [inline ecommerce/coupons/utils.py]: Note: No longer setting cache on cache hits.

File: ecommerce/courses/tests/test_utils.py [hunk 2]
```diff
@@ -73,8 +73,8 @@ def test_get_course_run_info_from_catalog(self, course_run):
         else:
             self.assertEqual(response['title'], product.title)
 
-        cached_course = cache.get(cache_key)
-        self.assertEqual(cached_course, response)
+        cached_course = TieredCache.get_cache_response(cache_key)
```
Comment (BenjiLee) [inline ecommerce/courses/tests/test_utils.py]: `get_cache_response`-> `get_cached_response`

Comment (BenjiLee) [reply ecommerce/courses/tests/test_utils.py]: `cached_course` -> `course_cached_response`

**Comment by BenjiLee** [comment]
@robrap Currently looking at the enterprise/api codcov

File: ecommerce/enterprise/tests/test_api.py [hunk 3]
```diff
@@ -145,6 +146,20 @@ def test_catalog_contains_course_runs(self, expected, enterprise_customer_catalo
...
+        Verify `catalog_contains_course_runs` returns a cached response
+        """
+        self.mock_catalog_contains_course_runs(
+            [self.course_run.id],
+            'fake-uuid',
+            enterprise_customer_catalog_uuid=None,
+            contains_content=True,
+        )
+        self._assert_contains_course_runs(True, [self.course_run.id], 'fake-uuid', None)
+        self._assert_contains_course_runs(True, [self.course_run.id], 'fake-uuid', None)
```
Comment (BenjiLee) [inline ecommerce/enterprise/tests/test_api.py]: @robrap Do you know of any good way to explicitly confirm a cache hit? 

This is the line this test is covering https://github.com/edx/ecommerce/blob/b621a2dc8317acdb0a1c8ea8ab68afec17aca162/ecommerce/enterprise/api.py#L192-L193

Comment (robrap) [reply ecommerce/enterprise/tests/test_api.py]: Not sure of the best way.  Some ideas:
1. Split method into _get_cache_X() and _get_X().  Then mock and check if _get_X() is called.
2. Spy on the method and see what it is doing.  Spying is when you let it do its thing, but you watch it do its thing.  At least in Jasmine that is what it is called I think.  Here is an example of spying I found with mocks:
http://wesmckinney.com/blog/spying-with-python-mocks/
You can see that it is setting "wraps" for the same thing you are patching.  Hopefully that would both make the real call and allow you to spy on its return value (somehow) and see that it returned is_hit.

File: ecommerce/enterprise/tests/test_api.py [hunk 4]
```diff
@@ -145,6 +146,19 @@ def test_catalog_contains_course_runs(self, expected, enterprise_customer_catalo
...
```
Comment (BenjiLee) [inline ecommerce/enterprise/tests/test_api.py]: Use the inner function `catalog_contains_course_runs`

File: ecommerce/courses/tests/test_views.py [hunk 5]
```diff
@@ -128,6 +130,27 @@ def test_credit_providers_in_context(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.context['credit_providers'], provider_json)
 
+    @httpretty.activate
+    def test_credit_providers_in_context_cached(self):
+
```
Comment (jmyatt) [inline ecommerce/courses/tests/test_views.py]: TODO: remove this extra whitespace line

**Review by BenjiLee — APPROVED** [review]

**Pull request merged**